src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke.main;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The Storage class is used to read and write data to and from the data file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Class constructor which specifies the file path where the Storage object will read/write data from/to.
+     *
+     * @param filePath The specific file path of the data file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Method which searches for the data file and reads each line of the file, converting it into an ArrayList of Task
+     * and returns it.
+     *
+     * @return ArrayList The ArrayList of tasks that was stored in the data file.
+     * @throws FileNotFoundException In the case that the file does not exist.
+     */
+    public ArrayList<Task> readPastData() throws FileNotFoundException {
```

Comment:
This method might be a little too long (>30 LOC), perhaps it might be better to include the cases in a separate method?